<launch>
<arg name="pi/2" value="1.5707963267948966" />
<arg name="optical_rotate" value="0 0 0 -$(arg pi/2) 0 -$(arg pi/2)" />
<node pkg="tf" type="static_transform_publisher" name="camera_base_link"
     args="$(arg optical_rotate) base_link front 100" />  



<!-- Run the ROS package stereo_image_proc -->
<group ns="/front" >
    <node pkg="stereo_image_proc" type="stereo_image_proc" name="stereo_image_proc"/>

    <!-- Odometry -->
    <node pkg="rtabmap_ros" type="stereo_odometry" name="stereo_odometry" output="screen">
        <remap from="left/image_rect"       to="left/image_rect"/>
        <remap from="right/image_rect"      to="right/image_rect"/>
        <remap from="left/camera_info"      to="left/camera_info"/>
        <remap from="right/camera_info"     to="right/camera_info"/>

        <param name="frame_id" type="string" value="base_link"/>
        <param name="odom_frame_id" type="string" value="odom"/>
        <param name="approx_sync" type="bool" value="false"/>
        <param name="queue_size" type="int" value="10"/>

        <param name="Odom/Strategy" value="1"/>
        <param name="Odom/GuessMotion" value="false"/>
        <param name="Odom/ResetCountdown" value="1"/>

        <param name="Vis/EstimationType" value="0"/>
        <param name="Vis/CorType" value="0"/>

        <param name="Odom/InlierDistance" type="string" value="0.1"/>
        <param name="Odom/MinInliers" type="string" value="10"/>
        <param name="Odom/RoiRatios" type="string" value="0.03 0.03 0.04 0.04"/>
        <param name="Odom/MaxDepth" type="string" value="10"/>
    </node>     
</group>

<group ns="rtabmap">   
  <node name="rtabmap" pkg="rtabmap_ros" type="rtabmap" output="screen" args="--delete_db_on_start">
     <param name="frame_id" type="string" value="base_link"/>
     <param name="subscribe_stereo" type="bool" value="true"/>
     <param name="subscribe_depth" type="bool" value="false"/>
     <param name="stereo_approx_sync" type="bool" value="true"/>

     <remap from="left/image_rect" to="/front/left/image_rect_color"/>
     <remap from="right/image_rect" to="/front/right/image_rect"/>
     <remap from="left/camera_info" to="/front/left/camera_info"/>
     <remap from="right/camera_info" to="/front/right/camera_info"/>

     <remap from="odom" to="/front/odom"/>

     <param name="queue_size" type="int" value="10"/>
     

     <remap from="grid_map" to="map"/>
     <!-- RTAB-Map's parameters -->
     <param name="Rtabmap/TimeThr" type="string" value="700"/>
     <param name="Rtabmap/DetectionRate" type="string" value="1"/>



     <param name="Stereo/MaxDisparity" value="1000"/>
     <param name="Stereo/OpticalFlow" value="false"/>



     <param name="Vis/MinInliers" type="string" value="10"/>
     <param name="Vis/InlierDistance" type="string" value="0.1"/>
  </node>
 </group>
</launch>
